### PR TITLE
create new conversation if current one is assigned to agent in case of fresh load

### PR DESCRIPTION
**Pull request by gargshantnu**
## What do you want to achieve?
- create new conversation if current one is assigned to agent in case of fresh load

## PR Checklist
- [x] I have tested it locally and all functionalities are working fine.

## How was the code tested?
- verified in local

### CODE REVIEW

File: webplugin/js/app/zendesk-chat-service.js [hunk 1]
```diff
@@ -137,3 +137,17 @@ function ZendeskChatService() {
        
     }
 };
+
+
+var onTabClickedHandlerForZendeskConversations = function (event) {
+    console.log("onTabClicked from zendesk: ", event, MCK_GROUP_MAP[event.tabId]);
+    if (kommunicate._globals.zendeskChatSdkKey) {
+        let currentGroupData = MCK_GROUP_MAP[event.tabId];
```
Comment (devashishmamgain) [inline webplugin/js/app/zendesk-chat-service.js:145]: please use var, let wouldn't work in some browsers

**Review by devashishmamgain — CHANGES_REQUESTED** [review]

File: webplugin/js/app/zendesk-chat-service.js (hunk 1 above)
Comment (gargshantnu) [reply webplugin/js/app/zendesk-chat-service.js]: done

File: webplugin/js/app/zendesk-chat-service.js [hunk 2]
```diff
@@ -137,3 +137,17 @@ function ZendeskChatService() {
        
     }
 };
+
+
+var onTabClickedHandlerForZendeskConversations = function (event) {
+    console.log("onTabClicked from zendesk: ", event, MCK_GROUP_MAP[event.tabId]);
+    if (kommunicate._globals.zendeskChatSdkKey) {
+        var currentGroupData = MCK_GROUP_MAP[event.tabId];
+        var assigneeInfo = currentGroupData && currentGroupData.users && Object.values(currentGroupData.users).find(member => {
```
Comment (devashishmamgain) [inline webplugin/js/app/zendesk-chat-service.js:146]: => {
will work?

File: webplugin/js/app/zendesk-chat-service.js [hunk 3]
```diff
@@ -137,3 +137,17 @@ function ZendeskChatService() {
...
+            return member.userId == currentGroupData.metadata.CONVERSATION_ASSIGNEE
+        })
+        if (assigneeInfo && assigneeInfo.role != 2) {
```
Comment (devashishmamgain) [inline webplugin/js/app/zendesk-chat-service.js:149]: any constant for 2?

**Review by devashishmamgain — CHANGES_REQUESTED** [review]

File: webplugin/js/app/zendesk-chat-service.js (hunk 3 above)
Comment (gargshantnu) [reply webplugin/js/app/zendesk-chat-service.js]: updated

File: webplugin/js/app/zendesk-chat-service.js (hunk 2 above)
Comment (gargshantnu) [reply webplugin/js/app/zendesk-chat-service.js]: updated

**Review by devashishmamgain — DISMISSED** [review]

**Review by Prochnost55 — DISMISSED** [review]

**Review by devashishmamgain — COMMENTED** [review]
what will happen if someone will click on back arrow and then open the old thread and send a message?

**Comment by devashishmamgain** [comment]
> what will happen if someone will click on back arrow and then open the old thread and send a message?

if it is shown, and doesn't work. are we showing any information to the user about it, or blocking the message sending itself?

**Comment by gargshantnu** [comment]
> what will happen if someone will click on back arrow and then open the old thread and send a message?

back arrow is hidden if zendesk chat is integrated.

**Review by devashishmamgain — APPROVED** [review]

**Pull request merged**